nautobot/extras/api/views.py

```diff
@@ -451,6 +450,67 @@ class JobModelViewSet(
     serializer_class = serializers.JobSerializer
     filterset_class = filters.JobFilterSet
 
+    # I don't see an easy way for swagger_auto_schema to provide a different schema depending on the API version.
+    # Since for now we default to the pre-1.3 API behavior, document that version in the schema for now.
+    @swagger_auto_schema(responses={"200": serializers.JobClassSerializer(many=True)})
+    def list(self, request, *args, **kwargs):
+        """List all known Jobs. Note that filtering of this list is only supported with API version 1.3 or later."""
+        if request.major_version > 1 or request.minor_version >= 3:
+            # API version 1.3 or later - standard model-based response
+            return super().list(request, *args, **kwargs)
+
+        # API version 1.2 or earlier - serialize JobClass records
+        if not request.user.has_perm("extras.view_job"):
+            raise PermissionDenied("This user does not have permission to view jobs.")
+        job_content_type = ContentType.objects.get(app_label="extras", model="job")
+        results = {
+            r.name: r
+            for r in JobResult.objects.filter(
+                obj_type=job_content_type,
+                status__in=JobResultStatusChoices.TERMINAL_STATE_CHOICES,
+            )
+            .defer("data")
+            .order_by("created")
+        }
+
+        job_models = Job.objects.restrict(request.user, "view")
+        jobs_list = [
+            job_model.job_class()  # TODO: why do we need to instantiate the job_class?
+            for job_model in job_models
+            if job_model.installed and job_model.job_class is not None
+        ]
+        for job_instance in jobs_list:
+            job_instance.result = results.get(job_instance.class_path, None)
+
+        serializer = serializers.JobClassSerializer(jobs_list, many=True, context={"request": request})
+
+        return Response(serializer.data)
+
+    @swagger_auto_schema(
+        responses={"200": serializers.JobClassDetailSerializer()}, operation_id="extras_jobs_read_deprecated"
+    )
+    @action(methods=["get"], detail=False, url_path="(?P<class_path>[^/]+/[^/]+/[^/]+)", url_name="detail")
+    def retrieve_deprecated(self, request, class_path):
```

Comment:
It's not clear how or where this is ever even being called?

Reply:
See the `url_path` on `@action` above - this is called when someone sends a GET to `/api/extras/jobs/<class_path>/`, i.e. the existing JobClass-based API detail view. I'll see about adding a docstring here to make that more obvious.